src/read/wasm.rs

```diff
@@ -254,25 +254,34 @@ impl<'data, R: ReadRef<'data>> WasmFile<'data, R> {
                         .get_name_section_reader()
                         .read_error("Couldn't read header of the name section")?
                     {
-                        let name =
-                            match name.read_error("Couldn't read header of a name subsection")? {
+                        // TODO: Right now, ill-formed name subsections
+                        // are silently ignored in order to maintain 
+                        // compatibility with extended name sections, which 
+                        // are not yet supported by the version of 
+                        // `wasmparser` currently used.
+                        // A better fix would be to update `wasmparser` to 
+                        // the newester version, but this requires
+                        // a major rewrite of this file.
+                        if let Ok(name) = name {
+                            let name = match name {
                                 wp::Name::Function(name) => name,
```

Comment:
Can you change this match to `Ok(wp::Name::Function(name))` instead of adding the `if let Ok(name) =name`?

Reply:
Done, thanks for the hint :+1:

Reply:
I meant to keep the `match name` instead of adding the `if let` (it minimizes changes and avoids the extra indentation), but going to merge because not worth fussing over.